docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-196.adoc

```diff
@@ -0,0 +1,60 @@
+== DOM updates should not lead to cross-site scripting (XSS) attacks
+
+=== Policy Details
+
+[width=45%]
+[cols="1,1"]
+|=== 
+|Prisma Cloud Policy ID 
+| 9743b4fb-4e52-484e-bc51-e8700e9f99aa
+
+|Checkov ID 
+|CKV3_SAST_196
+
+|Severity
+|HIGH
+
+|Subtype
+|Build
+
+|Language
+|JavaScript, TypeScript
+
+|CWEs
+|https://cwe.mitre.org/data/definitions/79.html[CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')]
+
+|OWASP Categories
+|https://owasp.org/Top10/A03_2021-Injection/[A03:2021 - Injection]
+
+|=== 
+
+=== Description
+
+This policy detects unsafe DOM updates that can lead to Cross-Site Scripting (XSS) attacks in JavaScript and TypeScript applications. XSS vulnerabilities occur when untrusted data is written to the DOM without proper validation or sanitization, allowing attackers to inject malicious scripts.
```

Comment:
This policy detects whether JavaScript and TypeScript applications perform unsafe DOM updates that can introduce Cross-Site Scripting (XSS) vulnerabilities. XSS vulnerabilities occur when untrusted data is written to the DOM without proper validation or sanitization, allowing attackers to inject malicious scripts.